src/event.rs

```diff
@@ -636,6 +636,11 @@ pub struct KeyboardInput {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum IME {
     /// Notifies when the IME was enabled.
+    ///
+    /// `KeyboardInput` and `ReceivedCharacter` events may still be sent
+    /// between the `IME::Enabled` and the `IME::Disabled` events.
+    /// The application may need to ignore some of those events while
+    /// the IME is enabled.
```

Comment:
So given that I'm not sure that comment wrt `Enabled` is relevant. Also on some X11 enabled is just send on start and send back on end, so it's mostly about ignoring events during preedit handling. Don't you think?

Reply:
Can you describe what `IME::Enabled` and `IME::Disabled` are used for as precisely as you can?

Reply:
> so it's mostly about ignoring events during preedit handling

Yes，I think you are right. But I still don't know what `Enabled` and `Disabled` are used for :(

Reply:
The original idea behind those two events were to indicate that the user has activated/deactivated some sort of input method. The `Enabled` would indicate that user should start performing calls to `window.set_ime_position`, the disabled would indicate that all pending IME handling could be safely aborted, since input method is gone.

Without those events the end user will have to always perform `set_ime_position` to keep IME popup in a meaningful place.

I think you're thinking of those events as a `Composition::Started` and `Composition::Ended`, but that's not the case, and those aren't required since `Preedit` and `Commit`/`Disabled` are some sort of indication that text composing started/ended.

Reply:
Now I understand. Thank you!

Reply:
But we should document that, because I bet that it won't be obvious to the users either.

Reply:
>Yes，I think you are right. But I still don't know what Enabled and Disabled are used for :(

They are just hints to not do any IME related processing when handling input. The thing is that this API is only about event, however there should be methods on a Window to provide a feedback wrt things you're doing, like setting surrounding text, etc, etc.

And if your IME is disabled you don't need to perform a series of `set_ime_position, set_surrounding_text, set_contnets_type` calls.

But the later calls are not here.

We could go completely without `Enabled` and `Disabled` events, and leave it up to winits' backends to optimize calls to mentioned methods when IME isn't present.

Reply:
>But we should document that, because I bet that it won't be obvious to the users either.

Yes, but I'll give a pass on documentation when we open it against master, since it'll likely be soon. We're waiting on X11 to be fixed and on macOS impl to be reviewed, since if backend doesn't support IME or the API isn't implemented it'll still work but via `ReceivedCharacter` API.